### ISSUE TITLE
Extension parsing requires slot number

### ISSUE THREAD

**Issue by SamuraiCrow** [AMOSFactory/AMOSProfessional#1]
Since Marc365 has apparently rage-quit and gone it alone, perhaps we should just address the minimum tasks and the first that comes to mind is that extensions are doomed to a particular slot number 1-26 and cannot be easily migrated to another slot.  I had proposed in the previous implementation that the token tables be replaced by a hash table.

**Comment by mness1978** [comment]
I agree, this is a pretty big issue and I think your solution has promise.  Did you even settle on a hash algorithm?  Have you studied BruceUncle's documentation on extensions and their token tables?

**Comment by SamuraiCrow** [comment]
The minimal perfect has algorithm was decided but I haven't studied BruceUncle's documentation.

**Comment by SamuraiCrow** [comment]
http://cmph.sourceforge.net/ and it's FCH algorithm was the decided hash maker.
